### PR TITLE
Feature/mat 3374 app client data reconstruction

### PR DESCRIPTION
**Pull request by luka-ethernal**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ authors = ["Polygon Avail Team"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-kate-recovery = {git = "https://github.com/maticnetwork/avail.git"}
+kate-recovery = {git = "https://github.com/maticnetwork/avail.git", branch="feature/MAT-3374-app-client-crate"}
```
Comment (prabal-banerjee) [inline Cargo.toml:10]: Can you post a link to the branch? Couldn't find it.

Comment (luka-ethernal) [reply Cargo.toml:10]: This branch has just been merged to main, I will revert this change as it is no longer needed.

**Review by prabal-banerjee — APPROVED** [review]

**Review by aterentic-ethernal — APPROVED** [review]

File: src/client.rs [hunk 2]
```diff
@@ -625,6 +659,38 @@ pub fn get_block_cid_entry(
...
+			layout.push((0, index[i].1 - 1))
+		} else if i == index.len() - 1 {
+			layout.push((index[i - 1].0, index[i].1 - index[i - 1].1));
+			layout.push((index[i].0, size - index[i].1 + 1));
+		} else {
+			layout.push((index[i - 1].0, index[i].1 - index[i - 1].1));
+		}
+	}
+	layout
+}
```
Comment (MiguelDD1) [inline src/client.rs:678]: - Move `if` evaluations conditions outside the loop.
-  `enumerate()` is not needed: 
- As we are not consuming `index`, we could just pass a reference to the slice.
```suggestion
fn layout_from_index(index: &[(u32, u32)], size: u32) -> Vec<(u32, u32)> {
	let mut layout: Vec<(u32, u32)> = Vec::with_capacity(index.len() + 1);
	if index.is_empty() {
		return vec![(0, size)];
	}

	layout.push((0, index[0].1 - 1));
	for i in 1..index.len() - 1 {
		layout.push((index[i - 1].0, index[i].1 - index[i - 1].1));
	}

	if index.len() > 1 {
		let last_idx = index.len() - 1;
		layout.push((
			index[last_idx - 1].0,
			index[last_idx].1 - index[last_idx - 1].1,
		));
		layout.push((index[last_idx].0, size - index[last_idx].1 + 1));
	}

	layout
}
```

**Review by MiguelDD1 — APPROVED** [review]

File: src/client.rs [hunk 3]
```diff
@@ -406,10 +407,44 @@ pub async fn run_client(
...
+								.enumerate()
+								.map(|(i, cells)| {
+									cells.clone().map(|data| Cell {
+										row: i as u16,
+										col: col_num,
+										data: data[48..].to_vec(),
+									})
+								})
+								.filter_map(identity)
+								.collect::<Vec<Cell>>()
```
[the remark this answers is not in the document]
Comment (luka-ethernal) [reply src/client.rs]: Thanks! I changed it.

**Pull request merged**